### ISSUE TITLE
Provide method to specify metric name or delimiter for collectd input

### ISSUE THREAD

**Issue by davidblewett** [influxdata/influxdb#1148]
Currently, the collectd input plugin takes the full name following this format: https://collectd.org/wiki/index.php/Naming_schema (or: host "/" plugin ["-" plugin instance] "/" type ["-" type instance] ). This ends up looking like:

```
my.host.name/cpu-0/cpu-idle
```

The / characters in the metric name make querying them unwieldy. For example, to match all CPUs, you'd have to use:

```
 /^my.host.name\\/cpu-[0-9]+\\/cpu-(user|idle)/i
```

I would suggest either making the generated metric name completely configurable or being able to specify an alternate delimiter. The former could be done via a format string, i.e.:

```
:host.:plugin-:plugin_instance.:type-:type_instance
```

Or, since all of the fields are included in the metric, you could do automatic aggregating multiple collectd metrics into a single metric in influxdb by host:

```
:plugin-:plugin_instance.:type-:type_instance
```

Or, to have say all CPU values in a single metric:

```
:host.:plugin.:type
```

The first example is not possible in the collectd aggregation plugin ( https://collectd.org/wiki/index.php/Plugin:Aggregation ) as the Host field is required.

**Comment by stanhu** [comment]
+1

**Comment by 3h4x** [comment]
I'm fighting with this right now
+1

**Comment by awmyhr** [comment]
Pulling my hair out over this
+1

**Comment by justin8** [comment]
+1. Really hard to use with grafana without this.

**Comment by h4wkmoon** [comment]
"Really hard to use with grafana without this."
so true.
+1

**Comment by ecki** [comment]
Hm, it looks like the metric is fully "tagyfied" in 0.9. There is no mentioning of this in this bug. Did I have dreamed it? :)

**Comment by beckettsean** [comment]
@jwilder are there filters for collectd like there are for graphite? If so, does that address this issue?

**Comment by faxm0dem** [comment]
It is possible to control what is being sent to which collectd *write* plugin using [*filter chains*](https://collectd.org/wiki/index.php/Chains). However, I don't think it's possible to send to one or another server using the wire protocol when multiple servers are configured. Maybe @mfournier @octo or @pyr can confirm this?

**Comment by octo** [comment]
@faxm0dem That depends on the write plugin being used. I assume InfluxDB is consuming the JSON data produced by the *write http plugin*? Then you can do:

```
Target write "write_http/<name>"
```

**Comment by faxm0dem** [comment]
no we're talking collectd network protocol

**Comment by octo** [comment]
No, with the *network plugin* that's unfortunately not possible :\

**Comment by bfgoodrich** [comment]
+1 - this would be a really nice feature

**Comment by druchoo** [comment]
+1

**Comment by russorat** [comment]
Has anyone tried solving this by putting telegraf in front of Influxdb? the socket listener input and the collectd parser should solve this i think.
https://github.com/influxdata/telegraf/tree/master/plugins/inputs/socket_listener
https://github.com/influxdata/telegraf/tree/master/plugins/parsers/collectd

**Comment by faxm0dem** [comment]
you can use [Riemann](/riemann/riemann) to munge the collectd metadata to your needs
